### PR TITLE
fix CLB bugs

### PR DESCRIPTION
**Pull request by gailwang**
fix bug that resource `tencent_clb_listener` 's unchangeable `health_check_switch`
fix bug that resource `tencent_clb_instance` read nil and report error
fix doc issues

### CODE REVIEW

**Comment by gailwang** [comment]
$ make testacc TESTARGS="-run=TestAccTencentCloudClb"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudClb -timeout 120m
?       github.com/terraform-providers/terraform-provider-tencentcloud  [no test files]
testing: warning: no tests to run
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/gendoc   (cached) [no tests to run]
=== RUN   TestAccTencentCloudClbInstancesDataSource_internal
=== PAUSE TestAccTencentCloudClbInstancesDataSource_internal
=== RUN   TestAccTencentCloudClbInstancesDataSource_open
=== PAUSE TestAccTencentCloudClbInstancesDataSource_open
=== RUN   TestAccTencentCloudClbListenerRulesDataSource
=== PAUSE TestAccTencentCloudClbListenerRulesDataSource
=== RUN   TestAccTencentCloudClbListenersDataSource
=== PAUSE TestAccTencentCloudClbListenersDataSource
=== RUN   TestAccTencentCloudClbRedirectionsDataSource
=== PAUSE TestAccTencentCloudClbRedirectionsDataSource
=== RUN   TestAccTencentCloudClbServerAttachment_tcp
--- PASS: TestAccTencentCloudClbServerAttachment_tcp (54.52s)
=== RUN   TestAccTencentCloudClbServerAttachment_http
--- PASS: TestAccTencentCloudClbServerAttachment_http (48.54s)
=== RUN   TestAccTencentCloudClbInstance_basic
=== PAUSE TestAccTencentCloudClbInstance_basic
=== RUN   TestAccTencentCloudClbInstance_open
=== PAUSE TestAccTencentCloudClbInstance_open
=== RUN   TestAccTencentCloudClbInstance_internal
=== PAUSE TestAccTencentCloudClbInstance_internal
=== RUN   TestAccTencentCloudClbListenerRule_basic
=== PAUSE TestAccTencentCloudClbListenerRule_basic
=== RUN   TestAccTencentCloudClbListenerRule_full
=== PAUSE TestAccTencentCloudClbListenerRule_full
=== RUN   TestAccTencentCloudClbListener_basic
=== PAUSE TestAccTencentCloudClbListener_basic
=== RUN   TestAccTencentCloudClbListener_tcp
=== PAUSE TestAccTencentCloudClbListener_tcp
=== RUN   TestAccTencentCloudClbListener_https
=== PAUSE TestAccTencentCloudClbListener_https
=== RUN   TestAccTencentCloudClbListener_tcpssl
=== PAUSE TestAccTencentCloudClbListener_tcpssl
=== RUN   TestAccTencentCloudClbRedirection_basic
--- PASS: TestAccTencentCloudClbRedirection_basic (70.55s)
=== RUN   TestAccTencentCloudClbRedirection_auto
--- PASS: TestAccTencentCloudClbRedirection_auto (66.76s)
=== CONT  TestAccTencentCloudClbInstancesDataSource_internal
=== CONT  TestAccTencentCloudClbListenerRule_basic
=== CONT  TestAccTencentCloudClbRedirectionsDataSource
=== CONT  TestAccTencentCloudClbListenerRulesDataSource
=== CONT  TestAccTencentCloudClbListenersDataSource
=== CONT  TestAccTencentCloudClbInstance_open
=== CONT  TestAccTencentCloudClbInstance_internal
=== CONT  TestAccTencentCloudClbInstance_basic
--- PASS: TestAccTencentCloudClbInstance_basic (35.63s)
=== CONT  TestAccTencentCloudClbListener_https
--- PASS: TestAccTencentCloudClbInstancesDataSource_internal (44.84s)
=== CONT  TestAccTencentCloudClbListener_tcpssl
--- PASS: TestAccTencentCloudClbInstance_open (122.56s)
=== CONT  TestAccTencentCloudClbListener_basic
--- PASS: TestAccTencentCloudClbInstance_internal (128.07s)
=== CONT  TestAccTencentCloudClbListener_tcp
--- PASS: TestAccTencentCloudClbListenersDataSource (137.88s)
=== CONT  TestAccTencentCloudClbListenerRule_full
--- PASS: TestAccTencentCloudClbListenerRule_basic (188.26s)
=== CONT  TestAccTencentCloudClbInstancesDataSource_open
--- PASS: TestAccTencentCloudClbListenerRulesDataSource (191.59s)
--- PASS: TestAccTencentCloudClbListener_basic (136.96s)
--- PASS: TestAccTencentCloudClbListener_https (228.49s)
--- PASS: TestAccTencentCloudClbListener_tcpssl (221.82s)
--- PASS: TestAccTencentCloudClbInstancesDataSource_open (88.45s)
--- PASS: TestAccTencentCloudClbRedirectionsDataSource (283.03s)
--- PASS: TestAccTencentCloudClbListener_tcp (159.57s)
--- PASS: TestAccTencentCloudClbListenerRule_full (172.69s)
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud     551.738s
?       github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/connectivity        [no test files]
?       github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud/ratelimit   [no test files]

File: tencentcloud/data_source_tc_clb_instances.go [hunk 1]
```diff
@@ -167,13 +168,13 @@ func dataSourceTencentCloudClbInstancesRead(d *schema.ResourceData, meta interfa
 	err := resource.Retry(readRetryTimeout, func() *resource.RetryError {
 		results, e := clbService.DescribeLoadBalancerByFilter(ctx, params)
 		if e != nil {
-			return retryError(e)
+			return retryError(errors.WithStack(e))
```
Comment (Sherlock-Holo) [inline tencentcloud/data_source_tc_clb_instances.go]: 尽量在 error 刚出现的地方就带上 stack 信息吧

File: tencentcloud/data_source_tc_clb_listeners.go [hunk 2]
```diff
@@ -175,13 +176,13 @@ func dataSourceTencentCloudClbListenersRead(d *schema.ResourceData, meta interfa
 	err := resource.Retry(readRetryTimeout, func() *resource.RetryError {
 		results, e := clbService.DescribeListenersByFilter(ctx, params)
 		if e != nil {
-			return retryError(e)
+			return retryError(errors.WithStack(e))
```
Comment (Sherlock-Holo) [inline tencentcloud/data_source_tc_clb_listeners.go]: 同上

File: tencentcloud/resource_tc_clb_listener_test.go [hunk 3]
```diff
@@ -97,7 +97,7 @@ func TestAccTencentCloudClbListener_https(t *testing.T) {
 					resource.TestCheckResourceAttr("tencentcloud_clb_listener.listener_https", "listener_name", "listener_https"),
 					resource.TestCheckResourceAttr("tencentcloud_clb_listener.listener_https", "port", "77"),
 					resource.TestCheckResourceAttr("tencentcloud_clb_listener.listener_https", "certificate_ssl_mode", "UNIDIRECTIONAL"),
-					resource.TestCheckResourceAttr("tencentcloud_clb_listener.listener_https", "certificate_id", "VjANRdz8"),
+					resource.TestCheckResourceAttr("tencentcloud_clb_listener.listener_https", "certificate_id", "VfqO4zkB"),
```
Comment (Sherlock-Holo) [inline tencentcloud/resource_tc_clb_listener_test.go]: `"VfqO4zkB"` 建议做成常量然后放进 `basic_test.go`

**Comment by gailwang** [comment]
$ make testacc TESTARGS="-run=TestAccTencentCloudClb"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudClb -timeout 120m
?       github.com/terraform-providers/terraform-provider-tencentcloud  [no test files]
testing: warning: no tests to run
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/gendoc   (cached) [no tests to run]
=== RUN   TestAccTencentCloudClbInstancesDataSource_internal
=== PAUSE TestAccTencentCloudClbInstancesDataSource_internal
=== RUN   TestAccTencentCloudClbInstancesDataSource_open
=== PAUSE TestAccTencentCloudClbInstancesDataSource_open
=== RUN   TestAccTencentCloudClbListenerRulesDataSource
=== PAUSE TestAccTencentCloudClbListenerRulesDataSource
=== RUN   TestAccTencentCloudClbListenersDataSource
=== PAUSE TestAccTencentCloudClbListenersDataSource
=== RUN   TestAccTencentCloudClbRedirectionsDataSource
=== PAUSE TestAccTencentCloudClbRedirectionsDataSource
=== RUN   TestAccTencentCloudClbServerAttachment_tcp
--- PASS: TestAccTencentCloudClbServerAttachment_tcp (49.87s)
=== RUN   TestAccTencentCloudClbServerAttachment_http
--- PASS: TestAccTencentCloudClbServerAttachment_http (46.71s)
=== RUN   TestAccTencentCloudClbInstance_basic
=== PAUSE TestAccTencentCloudClbInstance_basic
=== RUN   TestAccTencentCloudClbInstance_open
=== PAUSE TestAccTencentCloudClbInstance_open
=== RUN   TestAccTencentCloudClbInstance_internal
=== PAUSE TestAccTencentCloudClbInstance_internal
=== RUN   TestAccTencentCloudClbListenerRule_basic
=== PAUSE TestAccTencentCloudClbListenerRule_basic
=== RUN   TestAccTencentCloudClbListenerRule_full
=== PAUSE TestAccTencentCloudClbListenerRule_full
=== RUN   TestAccTencentCloudClbListener_basic
=== PAUSE TestAccTencentCloudClbListener_basic
=== RUN   TestAccTencentCloudClbListener_tcp
=== PAUSE TestAccTencentCloudClbListener_tcp
=== RUN   TestAccTencentCloudClbListener_https
=== PAUSE TestAccTencentCloudClbListener_https
=== RUN   TestAccTencentCloudClbListener_tcpssl
=== PAUSE TestAccTencentCloudClbListener_tcpssl
=== RUN   TestAccTencentCloudClbRedirection_basic
--- PASS: TestAccTencentCloudClbRedirection_basic (69.16s)
=== RUN   TestAccTencentCloudClbRedirection_auto
--- PASS: TestAccTencentCloudClbRedirection_auto (64.08s)
=== CONT  TestAccTencentCloudClbInstancesDataSource_internal
=== CONT  TestAccTencentCloudClbListenerRule_basic
=== CONT  TestAccTencentCloudClbInstance_basic
=== CONT  TestAccTencentCloudClbListenerRulesDataSource
=== CONT  TestAccTencentCloudClbListener_https
=== CONT  TestAccTencentCloudClbRedirectionsDataSource
=== CONT  TestAccTencentCloudClbInstance_internal
=== CONT  TestAccTencentCloudClbInstance_open
--- PASS: TestAccTencentCloudClbInstance_basic (30.58s)
=== CONT  TestAccTencentCloudClbInstancesDataSource_open
--- PASS: TestAccTencentCloudClbInstancesDataSource_internal (38.22s)
=== CONT  TestAccTencentCloudClbListenersDataSource
--- PASS: TestAccTencentCloudClbInstance_open (109.49s)
=== CONT  TestAccTencentCloudClbListener_basic
--- PASS: TestAccTencentCloudClbInstance_internal (111.91s)
=== CONT  TestAccTencentCloudClbListener_tcp
--- PASS: TestAccTencentCloudClbInstancesDataSource_open (98.37s)
=== CONT  TestAccTencentCloudClbListener_tcpssl
--- PASS: TestAccTencentCloudClbListener_https (169.88s)
=== CONT  TestAccTencentCloudClbListenerRule_full
--- PASS: TestAccTencentCloudClbListenerRule_basic (174.94s)
--- PASS: TestAccTencentCloudClbListenerRulesDataSource (177.22s)
--- PASS: TestAccTencentCloudClbListenersDataSource (175.24s)
--- PASS: TestAccTencentCloudClbListener_basic (113.94s)
--- PASS: TestAccTencentCloudClbRedirectionsDataSource (244.11s)
--- PASS: TestAccTencentCloudClbListener_tcp (134.43s)
--- PASS: TestAccTencentCloudClbListener_tcpssl (135.90s)
--- PASS: TestAccTencentCloudClbListenerRule_full (127.48s)
PASS

**Comment by gailwang** [comment]
#235

**Pull request merged**